keps/sig-instrumentation/2845-deprecate-klog-specific-flags-in-k8s-components/README.md

```diff
@@ -0,0 +1,239 @@
+
+```
+/usr/local/bin/kube-apiserver --log-file=/var/log/kube-apiserver.log
+```
+
+to
+
+```
+/go-runner --log-file=/var/log/kube-apiserver.log /usr/local/bin/kube-apiserver
+```
+
+### Caveats
+
+Is it ok for K8s components to drop support for subset of klog flags? 
+
+Technically K8s already doesn't support klog flags. Klog flags are renamed to 
+comply with K8s flag naming convention (underscores are replaced with hyphens).
+Full klog support was never promised to users and removal of those flags should
+be treated as removal of any other flag.
+
+Is it ok for K8s components to drop support writing to files?
+Writing directly to files is an important feature still used by users, but this
+doesn't directly necessitates direct support in components. By providing a
+external solution like go-runner we can allow community to develop more advanced
+features while maintaining high quality implementation within components.
+Having more extendable solution developed externally should be more beneficial
+to community when compared to forcing closed list of features on everyone.
+
+### Risks and Mitigations
+
+#### Users don't want to use go-runner as replacement.
+
+TODO
+
+#### Log processing in parent process causes performance problems
+
+TODO
+
+## Design Details
+
+TODO
+
+### Test Plan
+
+Go-runner is already used for scalability tests. We should ensure that we cover
+all existing klog features.
+
+### Graduation Criteria
+
+#### Alpha
+
+- Klog can be configured without registering flags
+- Kubernetes logging configuration drops global state
+- Go-runner is feature complementary to klog flags planned for deprecation
+- Projects in Kubernetes Org are migrated to go-runner
+
+#### Beta
+
+- Go-runner project is well maintained and documented 
+- Documentation on switching go-runner is publicly available
+- Kubernetes klog flags are marked as deprecated
+
+#### GA
+
+- Kubernetes klog flags are removed
+
+### Upgrade / Downgrade Strategy
+
+N/A
+
+### Version Skew Strategy
+
+N/A
+
+## Implementation History
+
+- 20/06/2021 - Original proposal created in https://github.com/kubernetes/kubernetes/issues/99270
+- 30/07/2021 - First KEP draft was created
+
+## Drawbacks
+
+Deprecating klog features outside klog might create confusion in community. 
+Large part of community doesn't know that klog was created from necessity and 
+is not the end goal for logging in Kubernetes. We should do due diligence to 
+let community know about our plans and their impact on external components 
+depending on klog.
+
+## Alternatives
+
+### Continue supporting all klog features 
+At some point we should migrate all logging 
+configuration to Options or Configuration. Doing so while supporting all klog
+features makes their future removal much harder.
+
+### Release klog 3.0 with removed features 
+Removal of those features cannot be done without whole k8s community instead of
+just k8s core components
```

Comment:
Maybe we can do this as a follow-up eventually...

Reply:
Based on problems with klog 2.0, I would prefer to accelerate migration to logr as it would make more sense strategically. We still would need some wrapper code to configure external library but not something worth our own library.